### PR TITLE
Document balanced tililing

### LINKED ISSUE #539: Documenting the "balanced OpenCL vs CPU tiling" option
### Rationale
On many machines we find pretty fast multicore cpus and a lot of system ram but a significantly smaller gpu memory with somewhat resrticted number crunching power. There is no problem using OpenCL code in modules as long as that doesn't lead to excessive tiling with large overlapping areas. Such a situation was rare, now we have as examples D&S or laplacian highlights which might behave bad and it would have been better to use cpu code instead.
************

In DT 4.2 we have the CL device specific conf key expanded by an **advantage** value, it is set to zero by default.

### How does this work
While processing the pixelpipe we estimate overall workloads for both OpenCL and CPU (this is **not** static but depends on parameters set in the module thus defining tile size, number of tiles, tile overlap ...). In most cases we would prefer the OpenCL code to be executed even if that would mean tiling as the graphics cards is mostly much faster (often 10 times faster if it is a dedicated card).

If the ratio of estimated workloads (CPU vs GPU) is larger than the **advantage** hint, dt will use the CPU for processing that specific module this time.

Unfortunately there is no way to use some sort of benchmark so far, you have to find out the GPU advantage for your system yourself. 

### How to set up the advantage hint?

1. Measure performance (-d perf) in laplacian highlights after setting "diameter of reconstruction" to something high.
2. While doing so make sure (by -d tiling) that the cl code doesn't tile.
3. Check for execution times with opencl on and off.
4. The advantage option can now be set to approx (cpu-time / gpu-time)

### PR DESCRIPTION
**Pull request by jenshannoschwalm**
- first try to document

Resolves #539

### CODE REVIEW

File: content/special-topics/mem-performance.md [hunk 1]
```diff
@@ -90,17 +90,27 @@ If you want to make maximal use of your GPU memory for OpenCL, you have three op
...
 - Set [preferences > processing > cpu / gpu / memory > tune OpenCL performance](../preferences-settings/processing.md#cpu--gpu--memory) to "memory size", which will use all of your device's memory, less a 400MB headroom. Please see the [section below](#id-specific-opencl-configuration) for other options related to this setting.
 
+## balanced OpenCL versus CPU tiling
+
+On many machines we find pretty fast multicore cpus with a lot of system ram but a significantly smaller gpu memory with somewhat restricted number crunching power (like typical integrated graphics). There is no problem using OpenCL code in modules as long as that doesn't lead to excessive tiling with large overlapping areas because of large requirements.
+In such cases it might be better to run untiled cpu code instead of heavy tiling on your gpu.
+
+While processing the pipeline we estimate overall workloads for both OpenCL and CPU (this is not static but depends on parameters set in the module thus defining the tile size, number of tiles and tile overlap.
```
Comment (gi-man) [inline content/special-topics/mem-performance.md:98]: This is missing a )

File: content/special-topics/mem-performance.md [hunk 2]
```diff
@@ -90,17 +90,27 @@ If you want to make maximal use of your GPU memory for OpenCL, you have three op
...
+
+In most cases we would prefer the OpenCL code to be executed even if that would mean tiling as the graphics cards is mostly much faster (often 10 times faster if it is a dedicated card).
+If the ratio of estimated workloads (CPU vs GPU) is larger than the **advantage** hint, dt will use the CPU for processing that specific module.
```
Comment (gi-man) [inline content/special-topics/mem-performance.md:101]: I would suggest using darktable instead of dt

**Comment by elstoc** [comment]
I've pushed a copy-edit. Please can you review.

**Comment by jenshannoschwalm** [comment]
Seems good to me.

**Comment by elstoc** [comment]
Thanks! Let's merge

**Pull request merged**